core/src/test/java/tech/tablesaw/TableAssertions.java

```diff
@@ -0,0 +1,49 @@
+package tech.tablesaw;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Arrays;
+import tech.tablesaw.api.Row;
+import tech.tablesaw.api.Table;
+
+public final class TableAssertions {
+
+  private TableAssertions() {}
+
+  public static void assertTableContents(Object[][] expected, Table actual) {
+
+    if (expected.length == 0) {
+      //empty table
+      assertEquals(0, actual.rowCount(), "Expected an empty table");
+      assertEquals(0, actual.columnCount(), "Expected a table with no columns");
+    } else {
+      assertEquals(
+          expected.length - 1, actual.rowCount(), "Table does not have the same number of rows");
+
+      assertArrayEquals(
```

Comment:
This would be really confusing if you called this method thinking that it was just asserting the table contents were equal to the given array

Reply:
Ok, it seems we have different mental pictures of what "contents" means. To me "contents" is "everything the table contains", including the headers, so I'd expect contents to include the headers. I would have called it assertRowValues if it was just the rows. 
I'll try to find a name that is less prone to misunderstandings and perhaps separate out the header assertions.

Reply:
I've changed my approach here, as I think I see what you were getting at initially. The initial Table is created using a column-wise approach - i.e. Column.create(), so if eyeballing both the old and new tables, it was hard to see the difference. 
I've added a convenience method that allows comparing in that column-wise fashion but without forcing the person to create the expected results as a Table. 
I think it would be easier to visualise the tables in the test if both were the other way around (row based construction) but I do agree that it is better to be consistent. 
Let me know what you think about this.